src/main/java/net/snowflake/client/core/ArrowSqlInput.java

```diff
@@ -0,0 +1,305 @@
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(
+              () -> converters.getBooleanConverter().getBoolean(value, columnType));
+        });
+  }
+
+  @Override
+  public byte readByte() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) ->
+            mapExceptions(() -> converters.getNumberConverter().getByte(value)));
+  }
+
+  @Override
+  public short readShort() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getShort(value, columnType));
+        });
+  }
+
+  @Override
+  public int readInt() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getInt(value, columnType));
+        });
+  }
+
+  @Override
+  public long readLong() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getLong(value, columnType));
+        });
+  }
+
+  @Override
+  public float readFloat() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getFloat(value, columnType));
+        });
+  }
+
+  @Override
+  public double readDouble() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(() -> converters.getNumberConverter().getDouble(value, columnType));
+        });
+  }
+
+  @Override
+  public BigDecimal readBigDecimal() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          return mapExceptions(
+              () -> converters.getNumberConverter().getBigDecimal(value, columnType));
+        });
+  }
+
+  @Override
+  public byte[] readBytes() throws SQLException {
+    return withNextValue(
+        (value, fieldMetadata) -> {
+          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+          int columnSubType = fieldMetadata.getType();
+          int scale = fieldMetadata.getScale();
+          return mapExceptions(
+              () -> {
+                if (value instanceof byte[]) {
+                  return (byte[]) value;
```

Comment:
Shoudln't this be a part of bytes converter?

Reply:
true, added this check to the converter.